XRTK-Core/Packages/com.xrtk.core/Definitions/DiagnosticsSystem/DiagnosticsDataProviderConfiguration.cs

```diff
@@ -0,0 +1,75 @@
+﻿// Copyright (c) XRTK. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using UnityEngine;
+using XRTK.Attributes;
+using XRTK.Definitions.Utilities;
+using XRTK.Interfaces.DiagnosticsSystem;
+
+namespace XRTK.Definitions.DiagnosticsSystem
+{
+    [Serializable]
+    public struct DiagnosticsDataProviderConfiguration
+    {
+        /// <summary>
+        /// Constructor.
+        /// </summary>
+        /// <param name="dataProviderType"></param>
+        /// <param name="dataProviderName"></param>
+        /// <param name="priority"></param>
+        /// <param name="runtimePlatform"></param>
+        /// <param name="profile"></param>
+        public DiagnosticsDataProviderConfiguration(SystemType dataProviderType, string dataProviderName, uint priority, SupportedPlatforms runtimePlatform, BaseMixedRealityProfile profile)
+        {
+            this.dataProviderType = dataProviderType;
+            this.dataProviderName = dataProviderName;
+            this.priority = priority;
+            this.runtimePlatform = runtimePlatform;
+            this.profile = profile;
+        }
+
+        [SerializeField]
+        [Tooltip("The concrete type of controller data provider to register.")]
+        [Implements(typeof(IMixedRealityDiagnosticsDataProvider), TypeGrouping.ByNamespaceFlat)]
+        private SystemType dataProviderType;
+
+        /// <summary>
+        /// The concrete type to use for this data provider.
+        /// </summary>
+        public SystemType DataProviderType => dataProviderType;
+
+        [SerializeField]
+        private string dataProviderName;
+
+        /// <summary>
+        /// The simple, human readable name for the system, feature, or manager.
+        /// </summary>
+        public string DataProviderName => dataProviderName;
+
+        [SerializeField]
+        private uint priority;
+
+        /// <summary>
+        /// The priority this system, feature, or manager will be initialized in.
+        /// </summary>
+        public uint Priority => priority;
+
+        [EnumFlags]
+        [SerializeField]
+        private SupportedPlatforms runtimePlatform;
+
+        /// <summary>
+        /// The runtime platform(s) to run this system, feature, or manager on.
+        /// </summary>
+        public SupportedPlatforms RuntimePlatform => runtimePlatform;
+
+        [SerializeField]
+        private BaseMixedRealityProfile profile;
```

Comment:
If we're not passing in and serializing a base diagnostics profile type then we probably don't need a configuration type just for the diagnostics data providers

Reply:
It might come in handy at some point. But I agree currently not used. The data provider configuration class is also needed to filter the options in in the inspector (`[Implements(typeof(IMixedRealityDiagnosticsDataProvider), TypeGrouping.ByNamespaceFlat)]`).

I can move that into the diagnostics system profile for sure but I am losing dataProviderName, priority and runtime platform options then as well and will have to reimplement that as well.

For me it seems having the configuration class is much less code required and more concistent with the rest.

Either that or I am not getting what you are suggesting. What's your proposed solution?

Reply:
I think what is now, we can reuse a type that already exists. But I see your appeal to a more refined capability in the future. 

I think leaving it in is best, but updating the `BaseMixedRealityProfile` to be the configuration profile for the diagnostics handlers you've got is all I'm really asking for.

Reply:
@StephenHodgson I updated it, can you check if this is what you want?